### ISSUE TITLE
Unable to resolve "kubernetes.default"

### ISSUE THREAD

**Issue by microadam** [projectcalico/k8s-policy#12]
Just tried to upgrade to v0.1.2 and all of a sudden I am getting:

```
2016-03-23 21:21:28,317 6 INFO Configuring /etc/hosts
2016-03-23 21:21:28,317 6 INFO Appended 'kubernetes.default  -> 10.233.0.1' to /etc/hosts
2016-03-23 21:21:28,317 6 INFO Beginning execution
2016-03-23 21:21:28,318 6 INFO Setting NetworkPolicy ADDED handler: <function add_update_network_policy at 0x7f26fe431050>
2016-03-23 21:21:28,318 6 INFO Setting NetworkPolicy DELETED handler: <function delete_network_policy at 0x7f27032f7c80>
2016-03-23 21:21:28,318 6 INFO Setting Namespace ADDED handler: <function add_update_namespace at 0x7f26fe43b848>
2016-03-23 21:21:28,318 6 INFO Setting Namespace DELETED handler: <function delete_namespace at 0x7f26fe43b938>
2016-03-23 21:21:28,318 6 INFO Setting Pod ADDED handler: <function add_update_pod at 0x7f26fe43ba28>
2016-03-23 21:21:28,319 6 INFO Setting Pod DELETED handler: <function delete_pod at 0x7f26fe43bb18>
2016-03-23 21:21:28,327 6 INFO Getting existing NetworkPolicy objects
2016-03-23 21:21:28,332 6 INFO Getting API resources 'https://kubernetes.default:443/apis/net.alpha.kubernetes.io/v1alpha1/networkpolicys' at version 'None'. stream=False
Traceback (most recent call last):
  File "<string>", line 337, in <module>
  File "<string>", line 109, in run
  File "<string>", line 126, in read_initial_state
  File "<string>", line 286, in _api_get
  File "site-packages/requests/sessions.py", line 480, in get
  File "site-packages/requests/sessions.py", line 468, in request
  File "site-packages/requests/sessions.py", line 576, in send
  File "site-packages/requests/adapters.py", line 437, in send
requests.exceptions.ConnectionError: HTTPSConnectionPool(host='kubernetes.default', port=443): Max retries exceeded with url: /apis/net.alpha.kubernetes.io/v1alpha1/networkpolicys (Caused by NewConnectionError('<requests.packages.urllib3.connection.VerifiedHTTPSConnection object at 0x7f26fe3d5dd0>: Failed to establish a new connection: [Errno -2] Name does not resolve',))
```

Completely baffled as to why, as if I run:

```
docker run --rm -it -e "K8S_SERVICE_HOST=10.233.0.1" -e "ETCD_ENDPOINTS=http://192.168.199.144:2379" calico/k8s-policy-agent:v0.1.2 /bin/sh
# echo -e "\n 10.233.0.1 kubernetes.default" >> /etc/hosts
# /dist/policy_agent
```
It can then connect, but then I get an SSL warning error...

Any thoughts?

**Comment by caseydavenport** [comment]
Hey @microadam - I'm looking into this right now.  We've changed the way we build the docker image and I believe we introduced a regression in the process.

I'll let you know when I get to the bottom of it - thanks for raising the issue!

**Comment by caseydavenport** [comment]
^ The SSL warning is probably because by running the container outside of k8s you don't get the proper cert mounted into the container by the kubelet.

**Comment by microadam** [comment]
Thanks casey! I assumed the latter. 

With regards to the regression, I downgraded to the previous version and had the same issue, so not entirely sure it's a regression. Let me know if I can be of any more assistance in debugging!

**Comment by caseydavenport** [comment]
Yeah, unfortunately the change we made to the build meant we had to rebuild versions v0.1.0 and v0.1.1.. so I believe the regression has been back-ported to those releases... sigh...

**Comment by caseydavenport** [comment]
OK - this seems to fix it:

https://github.com/projectcalico/k8s-policy/pull/13

After discussing with @tomdee, we think the new version of glibc we are using in the container made parsing `/etc/hosts` more sensitive to newlines.  I've tried a container with the above patch, and all seems to work :)

**Comment by caseydavenport** [comment]
@microadam - OK, I've merged the fix in #13 and I've triggered rebuilds of v0.1.1 and v0.1.2 to include the patch.

Could you re-pull v0.1.2 and make sure it works for you now before closing this issue?

**Comment by microadam** [comment]
@caseydavenport yup! that fixed it, thanks a lot!
